src/AppBundle/Subscriber/AutoLabelPRFromContentSubscriber.php

```diff
@@ -0,0 +1,108 @@
+<?php
+
+namespace AppBundle\Subscriber;
+
+use AppBundle\Event\GitHubEvent;
+use AppBundle\GitHubEvents;
+use AppBundle\Issues\GitHub\CachedLabelsApi;
+use Symfony\Component\EventDispatcher\EventSubscriberInterface;
+
+/**
+ * Looks at new pull requests and auto-labels based on text.
+ */
+class AutoLabelPRFromContentSubscriber implements EventSubscriberInterface
+{
+    private $labelsApi;
+
+    public function __construct(CachedLabelsApi $labelsApi)
+    {
+        $this->labelsApi = $labelsApi;
+    }
+
+    /**
+     * @param GitHubEvent $event
+     */
+    public function onPullRequest(GitHubEvent $event)
+    {
+        $data = $event->getData();
+        if ('opened' !== $action = $data['action']) {
+            $event->setResponseData(array('unsupported_action' => $action));
+
+            return;
+        }
+
+        $prNumber = $data['pull_request']['number'];
+        $prTitle = $data['pull_request']['title'];
+        $prBody = $data['pull_request']['body'];
+        $prLabels = array();
+
+        // the PR title usually contains one or more labels
+        foreach ($this->extractLabels($prTitle) as $label) {
+            $prLabels[] = $label;
+        }
+
+        // the PR body usually indicates if this is a Bug, Feature, BC Break or Deprecation
+        if (preg_match('/\|\s*Bug fix\?\s*\|\s*yes\s*/', $prBody, $matches)) {
+            $prLabels[] = 'Bug';
+        }
+        if (preg_match('/\|\s*New feature\?\s*\|\s*yes\s*/', $prBody, $matches)) {
+            $prLabels[] = 'Feature';
+        }
+        if (preg_match('/\|\s*BC breaks\?\s*\|\s*yes\s*/', $prBody, $matches)) {
+            $prLabels[] = 'BC Break';
+        }
+        if (preg_match('/\|\s*Deprecations\?\s*\|\s*yes\s*/', $prBody, $matches)) {
+            $prLabels[] = 'Deprecation';
+        }
+
+        $this->labelsApi->addIssueLabels($prNumber, $prLabels, $event->getRepository());
+
+        $event->setResponseData(array(
+            'pull_request' => $prNumber,
+            'pr_labels' => $prLabels,
+        ));
+    }
+
+    private function extractLabels($prTitle)
+    {
+        $labels = array();
+
+        // e.g. "[PropertyAccess] [RFC] [WIP] Allow custom methods on property accesses"
```

Comment:
what about automatically applying "Status: Needs Work" when the PR has the `[WIP]` label?